iOS support, Framework Support, and more

jerrymarino opened the pull request:
Create a iOS compiler invocation for iOS predicated by the playground
settings. It searches contents.xcplayground for iOS.

Additionally:

- Leverage the "Sources/" directory which is created by Xcode. The
playground convention is to put all files here. Setup linker flags and
rpaths for frameworks in Sources. Include all swift sources in Sources/.

- Derive a build directory based on the playground path.

fixes #1

jerrymarino committed:
iOS support, Framework Support, and more

Create a iOS compiler invocation for iOS predicated by the playground
settings. It searches contents.xcplayground for iOS.

Additionally:

- Leverage the "Sources/" directory which is created by Xcode. The
playground convention is to put all files here. Setup linker flags and
rpaths for frameworks in Sources. Include all swift sources in Sources/.

- Derive a build directory based on the playground path.

fixes #1

keith commented on runner.sh:
```diff
@@ -1,25 +1,85 @@
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks \
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks \
```
Might want to use `xcode-select -p` here instead so it works as users change xcode versions

jerrymarino replied:
Good call @keith. Thank you for the comment, I'll fix this today 👍

jerrymarino committed:
Runner iOS fixes

Use xcode-select to handle alt Xcode installations ( thanks @keith )
Upgrade target to iOS 8.0, since frameworks aren't supported in 7
